bootstrap.go

```diff
@@ -22,6 +23,8 @@ const (
 	ConnectionTypeSocket ConnectionType = iota
 	// ConnectionTypeHTTP sets the connection to Datadog to go over HTTP
 	ConnectionTypeHTTP
+	// ConnectionTypeAuto sets connection to HTTP or UNIX depending on supplied configuration of DD_TRACE_AGENT_URL
+	ConnectionTypeAuto
```

Comment:
Should we deprecate `ConnectionTypeSocket` and `ConnectionTypeHTTP`?

Reply:
Do we want anyone to be able to enforce this? Choice for the sake of choice?

Reply:
ok, fair

2 questions
* Does it make sense to depreciate immediately when a new feature is introduced?
* How does depreciation technically work here?

Reply:
When something is deprecated linters should trigger errors/warnings. To bump the dependency the error/warning can be resolved in 2 ways.

1. Migrate to the new implementation
2. Use linting exception comment to ignore it, but since deprecation should also mean freezing that part of the code one is choosing to be left behind.